user-registration-app/src/app/views/user-registration-view/user-registration-view.component.ts

```diff
@@ -71,130 +71,121 @@ export class UserRegistrationViewComponent implements OnInit {
+      .subscribe((registrations: RegistrationApiResponse[]) => {
+        if (registrations) {
+          this.activeRegistration = registrations.filter(registration => registration.hackathon.active)[0];
         }
+        regObservable.unsubscribe();
       });
+      // The error for registrations gets handled in user-profile-view
   }
 
   editAddressToggle() {
     this.willUpdateAddressFields = !this.willUpdateAddressFields;
   }
 
   getAddress(): string {
-    const noAddr = 'No address on file. Please add your address if you would like to receive some swag!'
-    if (this.registrations.length > 0) {
-      return (this.registrations[0].address ? this.registrations[0].address : noAddr);
-    } else {
-      return noAddr;
+    if (this.activeRegistration && this.activeRegistration.address) {
+      return this.activeRegistration.address;
     }
+    return 'No address on file. Please add your address if you would like to receive some swag!';
   }
 
-  attachNewAddress(reg: Registration): Registration {
-    const addrFields = this.updateAddressFields;
-    let newAddress = '';
-
-    for (const field in addrFields) {
-      if (addrFields[field]) {
-        newAddress += `${addrFields[field]}, `;
-      }
-    }
-
-    if (newAddress.slice(-2) === ', ') {
-      newAddress = newAddress.slice(0, -2);
-    }
-
-    reg.address = newAddress;
-    reg.shareAddressMlh = this.shareAddressMlh;
-    reg.shareAddressSponsors = this.shareAddressSponsors;
-    return reg
+  private consolidateAddress() {
+    return Object.values(this.updatedAddressFields).filter(field => field).join(', ');
   }
 
   submitAddress() {
-    this.progressService.start();
-
-    let reg = Registration.parseFromApiResponse(this.registrations[0]);
-    reg.hackathon = this.registrations[0].hackathon.uid;
-    reg = this.attachNewAddress(reg);
+    this.progressService.ref().start();
+    const reg = Registration.parseFromApiResponse(this.activeRegistration);
+    reg.address = this.consolidateAddress();
+    reg.shareAddressMlh = this.shareAddressMlh;
+    reg.shareAddressSponsors = this.shareAddressSponsors;
 
     this.httpService.submitAddress(reg)
-      .subscribe(() => {
-        this.progressService.complete();
-        this.alertsService.success('Your address has been updated. Please navigate away from the page to refresh this view.');
-      },         ({ error }) => {
-        this.alertsService.warning('Something may have gone wrong in that process. Contact a member of staff to check');
-      })
+      .subscribe(
+        () => {
+          this.progressService.ref().complete();
+          this.toastrService.success('Your address has been updated. Please navigate away from the page to refresh this view.');
+        },
+        ({ error }) => {
+          this.toastrService.warning('Something may have gone wrong in that process. Contact a member of staff to check');
+        })
   }
 
   submitClasses() {
```

Comment:
overall with this extra credit section... users find it very confusing cause after they select the class and reload the page there is no indication on the screen that the selection went through @rsquared226 this is another thing that should be added by someone

Reply:
I should add that there is no indication besides the popup/alert which is often not good enough for users lmao

Reply:
Facts ya

Reply:
We might be able to color the class to a different color to show it went through

Reply:
We could also have it checked by default